pkg/otelutil/otelutil.go

```diff
@@ -46,10 +50,20 @@ type TestingM interface {
 //	)
 //
 //	func TestMain(m *testing.M) {
-//		otelutil.TestMain(m)
+//		otelutil.TestMain(m, "integration-some-package-here")
 //	}
-func TestMain(m TestingM) {
-	cleanup, err := Setup()
+func TestMain(m TestingM, name string, onTypes ...testutil.TestType) {
```

Comment:
The idea behind `onTypes` is that they act as an allowlist for when to actually setup otel logging. If this is more than a single type, or it's empty (meaning, set up for everything), then we prefix the type of test being run on the file. This is because after some more testing, due to the way we `go test ./...` with all of our testing types, we wind up with 3 output files for each package, one which corresponds to each testing type. This makes us able to distinguish the testing types (as opposed to just subpackages) from each other as well as skip log output for test types that are empty for a given package (like the `acceptance` subpackage's unit and integration tests). If we opt-in to only one test type, as acceptance tests do, then no need to generate a prefix for the log output.